muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/dialogs/student/tabs/contact-events/editors/edit-event.tsx

```diff
@@ -0,0 +1,307 @@
+          .getLocalizedMoment(this.props.contactEvent.entryDate)
+          .toDate(),
+        text: this.props.contactEvent.text,
+        type: this.props.contactEvent.type,
+      },
+      props.contactEvent.id + "-edit-contact-event"
+    );
+  }
+
+  /**
+   * onCKEditorChange
+   * @param text text
+   */
+  onCKEditorChange = (text: string) => {
+    this.setStateAndStore(
+      { text },
+      this.props.contactEvent.id + "-edit-contact-event"
+    );
+  };
+
+  /**
+   * onDateChange
+   * @param date a Date
+   */
+  onDateChange = (date: Date) => {
+    this.setStateAndStore({ date: date }, this.props.contactEvent.id);
+  };
+
+  /**
+   * onTypeChange
+   * @param e event
+   */
+  onTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setStateAndStore(
+      { type: e.target.value as ContactTypes },
+      this.props.contactEvent.id + "-edit-contact-event"
+    );
+  };
+
+  /**
+   * clearUp
+   */
+  clearUp = () => {
+    this.setStateAndClear(
+      {
+        date: this.props.i18n.time
+          .getLocalizedMoment(this.props.contactEvent.entryDate)
+          .toDate(),
+        text: this.props.contactEvent.text,
+        type: this.props.contactEvent.type,
+      },
+      this.props.contactEvent.id + "-edit-contact-event"
+    );
+  };
+
+  /**
+   * editContactEvent
+   */
+  editContactEvent = () => {
+    this.setState({
+      locked: true,
+    });
+    this.props.editContactLogEvent(
+      this.props.studentUserEntityId,
+      this.props.contactEvent.id,
+      /**
+       * payload
+       */
+      {
+        creatorId: this.props.status.userId,
+        text: this.state.text,
+        entryDate: moment(this.state.date).format(),
+        type: this.state.type,
+      },
+      /**
+       * onSuccess
+       */
+      () => {
+        this.setState({
+          locked: false,
+        });
+        this.handleOnEditorClose();
+        this.clearUp();
+      }
+    );
+  };
+
+  /**
+   * handleCloseEditor
+   */
+  handleOnEditorClose = () => {
+    this.props.closeEditor && this.props.closeEditor();
+  };
+
+  /**
+   * render
+   * @returns JSX.Element
+   */
+  render() {
+    const editorTitle =
+      this.props.i18n.text.get("plugin.discussion.answertomessage.topic") +
+      " - " +
+      this.props.i18n.text.get("plugin.discussion.createmessage.content");
+
+    const content = (
+      <>
+        <div className="env-dialog__row env-dialog__row--new-contact-event">
+          <div className="env-dialog__form-element-container env-dialog__form-element-container--new-contact-event">
+            <label htmlFor="contactEventdate" className="env-dialog__label">
+              {this.props.i18n.text.get(
+                "plugin.guider.user.dialog.createContactEvent.date"
+              )}
+            </label>
+            <DatePicker
+              className="env-dialog__input"
+              id="contactEventdate"
+              onChange={(date: Date) => this.onDateChange(date)}
```

Comment:
Unnecessary callback. Just call this.onDateChange and its should work fine

Reply:
Cannot find name 'date'. Did you mean 'Date'?